### PR TITLE
fix docker build

### PR DESCRIPTION
**Pull request by sveitser**
Needed to compile openssl-sys crate.

CI: https://github.com/EspressoSystems/nitro-espresso-integration/actions/runs/11954000922

### CODE REVIEW

File: system_tests/full_challenge_impl_test.go [hunk 1]
```diff
@@ -206,6 +206,13 @@ func setupSequencerInboxStub(ctx context.Context, t *testing.T, l1Info *Blockcha
 		DelaySeconds:  big.NewInt(10000),
 		FutureSeconds: big.NewInt(10000),
 	}
+
+	//  Deploy EspressoTEEVerifier Mock
+	espressoTEEVerifierAddr, tx, _, err := mocksgen.DeployEspressoTEEVerifierMock(&txOpts, l1Client)
```
Comment (ImJeremyHe) [inline system_tests/full_challenge_impl_test.go:211]: Should this deployment be written in `deploy.go`?

Comment (Sneh1999) [reply system_tests/full_challenge_impl_test.go:211]: This is a test file, I think this should be fine.

File: system_tests/full_challenge_impl_test.go [hunk 2]
```diff
@@ -215,10 +222,12 @@ func setupSequencerInboxStub(ctx context.Context, t *testing.T, l1Info *Blockcha
 		big.NewInt(117964),
 		reader4844,
 		false,
+		espressoTEEVerifierAddr,
 	)
 	Require(t, err)
 	_, err = EnsureTxSucceeded(ctx, l1Client, tx)
 	Require(t, err)
+
```
Comment (ImJeremyHe) [inline system_tests/full_challenge_impl_test.go:230]: LGTM. Only that I think we should minimize our changes on nitro/celestia's files, which is good for syncing

**Review by ImJeremyHe — APPROVED** [review]

File: .github/workflows/espresso-docker.yml [hunk 3]
```diff
@@ -158,60 +158,61 @@ jobs:
...
-            ls -lah
-            exit 1
-          fi
-
-          docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
-            $(printf '${{ matrix.image }}@sha256:%s ' *)
-      - name: Inspect image
-        run: |
-          docker buildx imagetools inspect ${{ matrix.image }}:${{ steps.meta.outputs.version }}
+  # Only building for AMD64
```
Comment (sveitser) [inline .github/workflows/espresso-docker.yml:161]: @Sneh1999 why comment this out? The arm image is used by anyone developing on a mac.

Comment (sveitser) [reply .github/workflows/espresso-docker.yml:163]: So I tried re-enabling it but now I'm thinking the part of extracting the wavm binary we may not need anymore as long as we always use the upstream one.

**Review by Sneh1999 — COMMENTED** [review]
Thanks @ImJeremyHe  and @sveitser  for all your help!
Will merge this now

**Pull request merged**